.github/actions/enforce-pr-conventions/action.yml

```diff
@@ -0,0 +1,70 @@
+name: enforce-pr-conventions
+
+description: Check if the branch name and pull request title follow conventional checks.
+
+inputs:
+  jira-project-name:
+    description: 'The JIRA project name.'
+    required: false
+  valid-branch-regex:
+    description: 'The regular expression to check the branch name.'
+    required: false
+  valid-pr-title-regex:
+    description: 'The regular expression to check the pull request title.'
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Get branch name
+      if: github.event_name == 'pull_request'
+      uses: Alfresco/alfresco-build-tools/.github/actions/get-branch-name@v5.13.1
+
+    - name: Generate regex
+      shell: bash
+      if: github.event_name == 'pull_request'
+      id: generate-regex
+      run: |
+        if [[ -z "${{ inputs.valid-branch-regex }}" ]]; then
+          echo "valid-branch-regex=^(improvement|bug|feature|test|tmp)/${{ inputs.jira-project-name }}-[0-9]+-[A-Za-z0-9._-]+$" >> $GITHUB_OUTPUT
+        else
+          echo "valid-branch-regex=${{ inputs.valid-branch-regex }}" >> $GITHUB_OUTPUT
+        fi
```

Comment:
you need to handle the case where the user do not provide either `jira-project-name` or `valid-branch-regex` or it will fail in a very confusing way?

Reply:
Added